ofctrl/ofctrl.go

```diff
@@ -28,11 +28,51 @@ import (
 	"antrea.io/libOpenflow/common"
 	"antrea.io/libOpenflow/openflow15"
 	"antrea.io/libOpenflow/util"
-
 	log "github.com/sirupsen/logrus"
 )
 
-type PacketIn openflow15.PacketIn
+type PacketIn struct {
+	*openflow15.PacketIn
+	UserData     []byte
+	Continuation []byte
+}
+
+func parsePacktInFromNXPacketIn2(pktIn2 *openflow15.PacketIn2) *PacketIn {
+	pktIn := openflow15.NewPacketIn()
+	var userData, continuation []byte
+	for _, p := range pktIn2.Props {
+		switch v := p.(type) {
+		case *openflow15.PacketIn2PropPacket:
+			pktBytes, _ := v.Packet.MarshalBinary()
+			pktIn.Data = util.NewBuffer(pktBytes)
+		case *openflow15.PacketIn2PropCookie:
+			pktIn.Cookie = v.Cookie
+		case *openflow15.PacketIn2PropBufferID:
+			pktIn.BufferId = v.BufferID
+		case *openflow15.PacketIn2PropReason:
+			pktIn.Reason = v.Reason
+		case *openflow15.PacketIn2PropFullLen:
+			pktIn.TotalLen = uint16(v.FullLen)
+		case *openflow15.PacketIn2PropTableID:
+			pktIn.TableId = v.TableID
+		case *openflow15.PacketIn2PropMetadata:
+			pktIn.Match = openflow15.Match{
+				Type:   openflow15.MatchType_OXM,
+				Length: v.Len(),
+				Fields: v.Fields,
+			}
+		case *openflow15.PacketIn2PropUserdata:
+			userData = v.Userdata
+		case *openflow15.PacketIn2PropContinuation:
+			continuation = v.Continuation
```

Comment:
Is there any chance that an unsupported type is passed here?

Reply:
I don't think so as the parameter of this function is type `PacketIn` (not raw string), all kinds of property type should be known.

Reply:
Make sense.

Reply:
Nit: Since there are lot of properties, adding a default case would help to catch any bug if any property is missed or for any bug in libOpenflow, otherwise it can stay latent.